mlrun/projects/project.py

```diff
@@ -3085,6 +3128,40 @@ def remove_custom_packager(self, packager: str):
         """
         self.spec.remove_custom_packager(packager=packager)
 
+    def _run_authenticated_git_action(
```

Comment:
lets add a unit test for this. as part of the test you should verify that the clean url is set again.

Reply:
Good call.

I added a couple of tests for us. I'm just not sure if they are 100% compliant with the expected patterns for our tests. Feel free to make suggestions there as well, please.